src/imap/mod.rs

```diff
@@ -575,6 +575,16 @@ impl Imap {
                 // new messages is only one command, just as a SELECT command)
                 true
             } else if let Some(uid_next) = mailbox.uid_next {
+                if uid_next < old_uid_next {
+                    // See https://github.com/deltachat/deltachat-core-rust/issues/2188
```

Comment:
I don't think this kind of comments is useful. This depends on GitHub and deltachat-core repository to be up. And even then, you have to follow the link and read the issue with all the discussions there. Either summarize or just remove the comment, following `warn!` is enough to understand what is going on.